pmd-core/src/main/resources/cpd-report_1_0_0.xsd

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0"?>
+<xs:schema
+        xmlns:xs="http://www.w3.org/2001/XMLSchema"
+        xmlns="https://pmd-code.org/ns/cpd-report/1.0.0"
+        targetNamespace="https://pmd-code.org/ns/cpd-report/1.0.0"
+        elementFormDefault="qualified">
+
+    <xs:element name="pmd-cpd">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element name="file" type="file" minOccurs="0" maxOccurs="unbounded"/>
+                <xs:element name="duplication" type="duplication" minOccurs="0" maxOccurs="unbounded"/>
+                <xs:element name="error" type="error" minOccurs="0" maxOccurs="unbounded"/>
+            </xs:sequence>
+            <xs:attribute name="version" type="xs:string" use="required"/>
+            <xs:attribute name="timestamp" type="xs:string" use="required"/>
+        </xs:complexType>
+    </xs:element>
+
+    <xs:complexType name="file">
+        <xs:attribute name="path" type="xs:string" use="required"/>
+        <xs:attribute name="totalNumberOfTokens" type="xs:int" use="required"/>
+    </xs:complexType>
+
+    <xs:complexType name="duplication">
+        <xs:sequence>
+            <xs:element name="file" type="fileLocation" minOccurs="1" maxOccurs="unbounded"/>
+            <xs:element name="codefragment" type="codefragment" minOccurs="1" maxOccurs="1"/>
+        </xs:sequence>
+        <xs:attribute name="lines" type="xs:positiveInteger" use="required"/>
+        <xs:attribute name="tokens" type="xs:positiveInteger" use="required"/>
+    </xs:complexType>
+
+    <xs:complexType name="error">
+        <xs:simpleContent>
+            <xs:extension base="xs:string">
+                <xs:attribute name="filename" type="xs:string" use="required"/>
+                <xs:attribute name="msg" type="xs:string" use="required"/>
+            </xs:extension>
+        </xs:simpleContent>
+    </xs:complexType>
+
+    <xs:complexType name="fileLocation">
+        <xs:attribute name="column" type="xs:positiveInteger" use="required"/>
+        <xs:attribute name="endcolumn" type="xs:positiveInteger" use="required"/>
+        <xs:attribute name="endline" type="xs:positiveInteger" use="required"/>
+        <xs:attribute name="line" type="xs:positiveInteger" use="required"/>
+        <xs:attribute name="begintoken" type="xs:integer" use="required"/>
+        <xs:attribute name="endtoken" type="xs:integer" use="required"/>
```

Comment:
```suggestion
        <xs:attribute name="begintoken" type="xs:positiveInteger" use="required"/>
        <xs:attribute name="endtoken" type="xs:positiveInteger" use="required"/>
```

or can these be 0? if so it should still be `xs:nonNegativeInteger` rather than `xs:integer`

Reply:
They can be zero, the first token of a file